Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Views/AssetBrowserTreeView.cpp

```diff
@@ -590,86 +590,92 @@ namespace AzToolsFramework
-                                msgBox.setText("The asset you are moving may be referenced in other assets.");
-                                msgBox.setInformativeText("More information can be found by pressing \"Show Details...\".");
-                                auto* moveButton = msgBox.addButton("Move", QMessageBox::YesRole);
-                                msgBox.setStandardButtons(QMessageBox::Cancel);
-                                msgBox.setDefaultButton(QMessageBox::Yes);
-                                msgBox.setDetailedText(message.c_str());
-                                QSpacerItem* horizontalSpacer = new QSpacerItem(600, 0, QSizePolicy::Minimum, QSizePolicy::Expanding);
-                                auto* layout = qobject_cast<QGridLayout*>(msgBox.layout());
-                                layout->addItem(horizontalSpacer, layout->rowCount(), 0, 1, layout->columnCount());
-                                msgBox.exec();
-
-                                if (msgBox.clickedButton() == reinterpret_cast<QAbstractButton *>(moveButton))
+                                if (message.size())
                                 {
-                                    AssetChangeReportRequest moveRequest(
-                                        AZ::OSString(fromPath.c_str()),
-                                        AZ::OSString(toPath.c_str()),
-                                        AssetChangeReportRequest::ChangeType::Move);
-                                    AssetChangeReportResponse moveResponse;
-                                    if (SendRequest(moveRequest, moveResponse))
+                                    QMessageBox msgBox(this);
+                                    msgBox.setWindowTitle("Before Move Asset Information");
+                                    msgBox.setIcon(QMessageBox::Warning);
+                                    msgBox.setText("The asset you are moving may be referenced in other assets.");
+                                    msgBox.setInformativeText("More information can be found by pressing \"Show Details...\".");
+                                    auto* moveButton = msgBox.addButton("Move", QMessageBox::YesRole);
+                                    msgBox.setStandardButtons(QMessageBox::Cancel);
+                                    msgBox.setDefaultButton(QMessageBox::Yes);
+                                    msgBox.setDetailedText(message.c_str());
+                                    QSpacerItem* horizontalSpacer = new QSpacerItem(600, 0, QSizePolicy::Minimum, QSizePolicy::Expanding);
+                                    auto* layout = qobject_cast<QGridLayout*>(msgBox.layout());
+                                    layout->addItem(horizontalSpacer, layout->rowCount(), 0, 1, layout->columnCount());
+                                    msgBox.exec();
+
+                                    if (msgBox.clickedButton() == reinterpret_cast<QAbstractButton*>(moveButton))
```

Comment:
why reinterpret_cast you can do a `staticcast`

Reply:
+1

Reply:
error C2440: 'static_cast': cannot convert from 'QPushButton *' to 'QAbstractButton *'
error C2440:                                     if (msgBox.clickedButton() == static_cast<QAbstractButton*>(moveButton))
error C2440:                                        ^
message : Types pointed to are unrelated; conversion requires reinterpret_cast, C-style cast or function-style cast
message :                                     if (msgBox.clickedButton() == static_cast<QAbstractButton*>(moveButton))
message :                                                                   ^

Reply:
Did you include the qpushbutton hear.

Reply:
error C2440: 'static_cast': cannot convert from 'QAbstractButton *' to 'QPushButton *'
error C2440:                                     if (static_cast<QPushButton *>(msgBox.clickedButton()) != moveButton)
error C2440:                                        ^
message : Types pointed to are unrelated; conversion requires reinterpret_cast, C-style cast or function-style cast
message :                                     if (static_cast<QPushButton *>(msgBox.clickedButton()) != moveButton)
message :                                         ^

Reply:
Doh!!!